src/java.base/windows/native/libnet/NetworkInterface_winXP.c

```diff
@@ -274,20 +268,23 @@ int getAllInterfacesAndAddresses (JNIEnv *env, netif **netifPP)
     // Retrieve IPv4 addresses with the IP Helper API
     curr = *netifPP;
     ret = lookupIPAddrTable(env, &tableP);
-    if (ret < 0) {
+    if (ret == -1) {
+      free_netif(*netifPP);
       return -1;
+    } else if (ret == -2) {
+        // Clear the exception and continue.
+        if ((*env)->ExceptionCheck(env)) {
+            (*env)->ExceptionClear(env);
+        }
+        tableP = NULL;
     }
     while (curr != NULL) {
         netaddr *netaddrP;
```

Comment:
netaddr *netaddrP = NULL;  just in case enumAddresses_win_ipaddrtable does not update it for an error return

Reply:
netaddrP is always updated when `enumAddresses_win_ipaddrtable` returns a nonnegative value